### PR TITLE
Add Dj-rest-auth

### PR DESCRIPTION
**Pull request by tincke10**
This pull request adds Dj-rest-auth to the project, which provides authentication and registration endpoints for the API.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+export FACEBOOK_SECRET=secret-de-facebook
+
+
+EMAIL_BACKEND = "django.core.mail.backends.smtp.EmailBackend"
+EMAIL_HOST = "<your email host>"                    # smtp-relay.sendinblue.com
+EMAIL_USE_TLS = False                               # False
+EMAIL_PORT = "<your email port>"                    # 587
+EMAIL_HOST_USER = "<your email user>"               # your email address
+EMAIL_HOST_PASSWORD = "<your email password>"       # your password
+DEFAULT_FROM_EMAIL = "<your default from email>"    # email ending with @sendinblue.com
```
Comment (edvm) [inline .env.example]:
```suggestion
DEFAULT_FROM_EMAIL = "<your default from email>"    # email from domain
```

File: app/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+    ]
+
+    operations = [
```
Comment (edvm) [inline app/migrations/0001_initial.py:13]: Why do we have these 'dummy' models here? :)

File: backend/settings.py [hunk 3]
```diff
@@ -89,26 +104,68 @@
...
+ACCOUNT_EMAIL_REQUIRED = True
+ACCOUNT_EMAIL_VERIFICATION = "mandatory"
+ROOT_URLCONF = "backend.urls"
+
+
+# URL base para redireccionar la confirmación de correo electrónico
+EMAIL_CONFIRM_REDIRECT_BASE_URL = "http://localhost:3000/email/confirm/"
+
+# URL base para redireccionar el restablecimiento de contraseña
+PASSWORD_RESET_CONFIRM_REDIRECT_BASE_URL = "http://localhost:3000/password-reset/confirm/"
```
Comment (edvm) [inline backend/settings.py]: CORS url is like:
image
And here its hardcoded to host:3000. Could this be a problem ?for example when sending a POST to reset password?

File: app/migrations/0001_initial.py (hunk 2 above)
Comment (gascarcella) [reply app/migrations/0001_initial.py:13]: This is automatic by Django

File: backend/settings.py [hunk 4]
```diff
@@ -89,26 +104,68 @@
...
+
+SITE_ID = 1
+
+ACCOUNT_EMAIL_REQUIRED = True
+ACCOUNT_EMAIL_VERIFICATION = "mandatory"
+ROOT_URLCONF = "backend.urls"
+
+
+# URL base para redireccionar la confirmación de correo electrónico
+EMAIL_CONFIRM_REDIRECT_BASE_URL = "http://localhost:3000/email/confirm/"
```
Comment (gascarcella) [inline backend/settings.py]: Let's use ENV vars for this.

**Review by gascarcella — COMMENTED** [review]
Looks good! Just a few comments

**Pull request closed without merging**